json-rpc/src/stream_rpc/errors.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{error, fmt};
+
+#[derive(Debug)]
+pub enum Error {
+    CouldNotStringifyMessage(String),
+    ClientAlreadyClosed(u64),
+    ClientWantsToDisconnect,
+    TransportError(String),
+}
+
+impl Error {
+    fn message(&self) -> String {
+        match self {
+            Error::CouldNotStringifyMessage(s) => format!("Could not convert {} to string", s),
+            Error::ClientAlreadyClosed(client_id) => format!("Client#{} is closed", client_id),
+            Error::ClientWantsToDisconnect => {
+                "Received disconnect request message from client".to_string()
+            }
+            Error::TransportError(s) => format!("Transport error: {}", s),
```

Comment:
I'm a fan of [thiserror](https://github.com/dtolnay/thiserror#example). It'll allow you to decorate the enum variations above and not have to explicitly create these `fn message/Display`s, which would make the code more maintainable.

https://github.com/diem/diem/blob/7555d8177ee4a87e2bfcc0ffcb1f7b9139be326c/sdk/offchain/src/identifier.rs#L122-L134

Reply:
Nice! This is really neat